core/src/main/java/org/apache/calcite/rel/rules/SortRemoveRedundantRule.java

```diff
@@ -134,7 +135,7 @@ private static Optional<BigDecimal> getRowCountThreshold(Sort sort) {
     if (RelOptUtil.isLimit(sort)) {
       final BigDecimal fetch =
           sort.fetch instanceof RexLiteral
-              ? RexLiteral.roundedValue(sort.fetch)
+              ? (BigDecimal) RexLiteral.numberValue(sort.fetch)
               : BigDecimal.ZERO;
```

Comment:
`((RexLiteral) sort.fetch).getValueAs(BigDecimal.class)` makes fewer assumptions

Reply:
Indeed, it can only be RexLiteral. Thanks!